### DISCUSSION TITLE
Commercialization

### DISCUSSION THREAD

**Discussion by erlend-sh** [muni-town/weird#14] [General]
[Latest draft here.](https://github.com/commune-os/weird/discussions/14#discussioncomment-9221264)

- - -

There will be two tiers of Weird service.

- plain Weird 🆓 
- Weird plus 💰 

### plain Weird

..will always be free. It consists of two things:

- Website generation.
- OIDC profile (Rauthy).

*However*, one can only attain membership by invitation from another member. This is to ensure a baseline of user authenticity. If someone invites a member that turns out to be a spambot, the inviter will also be heavily scrutinized. Paying for an account is the easiest way to authenticate the member as well-intentioned, so this invitation system serves as another such vetting system.

### Weird plus

Costs $5/month; first 100 customers can pay $25 for a year.
Enables custom domains and includes a core service add-on:

- Link directory ([linkblocks](https://github.com/raffomania/linkblocks))
- Custom domain

unlock any additional service for $2 per add-on. $1 each when surpassing $11/month.

- ActivityPub account (Kitsune? Mitra? GoToSocial?)
- ATproto account (Bluesky)
- Matrix account (Conduit)
- [Polyproto](https://docs.polyphony.chat/Protocol%20Specifications/core/) account
- Email forwarding (Stalwart)
- Bookmarks manager ([Shiori](https://github.com/go-shiori/shiori))
- RSS reader ([Miniflux](https://github.com/miniflux/v2/) / [Collie](https://github.com/parksb/collie)), [Feedi](https://github.com/facundoolano/feedi)
- Pastebin ([opengist](https://github.com/thomiceli/opengist)) or [Memos](https://github.com/usememos/memos)
- Blog (?)

#### Revenue share

Every open source service included in the Weird bundle will be offered $0.50 for each paying user of that service in our platform. The mechanics of this split are likely to change as the platform matures and we learn more about our [unit economics](https://www.paddle.com/resources/unit-economics) and margins. We’ll hold ourselves responsible to pay a fair share forward by keeping our revenue metrics transparent.

---

### Subsidizing freemium memberships

[Free plans are commercially dangerous](https://baremetrics.com/blog/freemium-saas-implode). Oftentimes they only make sense for a VC-funded business that can afford to break into a new market at a loss. We're not planning to do any of that.

At the same time, a free service is a more accessible service, so we want to figure out a way to sustainably provide 0$ premium accounts to anyone who'd need one.

The basic plan is to do something like a 30/70 split between paid/free users. Basically the number of paid users we have corresponds with the number of freemium slots we have available; if there aren't enough paying users, registrants for freemium will be waiting in queue.

To begin with it'll probably be more like 10/90 paid/free. It all comes down to sustainable operating margins, and we will be transparent with our community about what those look like for us.

Our model is inspired by the likes of the [Waking Up](https://www.wakingup.com/) meditation app, which is a highly priced premium app, but they have a simple policy of ‘anyone who emails us directly and asks for it will get a one year subscription, and you can ask anew forever’.

- - -

Prior art:
- https://github.com/directus/directus/discussions/17977
- https://github.com/usebruno/bruno/discussions/269

**Comment by zicklag** [comment]
> To begin with it'll probably be more like 10/90 paid/free. It all comes down to sustainable operating margins, and we will be transparent with our community about what those look like for us.

The good news is that with our server requirements, I think we might be able to handle thousands of users, even on a tiny $5/month server.

The free plan has a hard-cap on the amount of data stored per user ( if we put reasonable link and bio size limits ), too, so while we'll still have to monitor the impact of free accounts on the server, and how much bandwidth we're using, I think we'll be able to support many free accounts without having to worry too much about costs.

**Comment by anderspitman** [comment]
This is a very interesting business model. I love the idea, but I wonder if you'll be able to support enough free users in the long run. I think typical ratios for freemium services are much higher than 30/70 or even 10/90. Definitely worth trying though.

**Reply by erlend-sh** [reply]
When free is the *default*, the ratio is indeed more like 1% to 0.1% of users. We’re not doing the same as e.g. [Discord Nitro](https://discord.com/blog/beginners-guide-to-nitro) though, in that the only thing we’re giving away for free is the ID+website service that costs practically nothing for us to host. That’s more like if basic Discord was read-only.

To *do* just about anything, you’ll need Weird+. But because we don’t want to make the interactive services exclusively pay-to-play, we are going for a 30/70 ratio *in the paid plan*. That ratio won’t change unexpectedly, because the amount of free Weird+ accounts available is literally hard capped:
> if there aren't enough paying users, registrants for freemium will be waiting in queue.

**Comment by erlend-sh** [comment]
### Alternative payment scheme

The first scheme I came up with looks like this:

> Weird plus — Costs $5/month; first 100 customers can pay $25 for a year.
> Enables custom domains and includes a core service add-on: Link directory
> Unlock any additional service for $2 per add-on. $1 each when surpassing $11/month.

What I liked about the tiered unlocks:
- implicitly teaches users that running services cost money; more apps = higher price
- makes the rev-share math very simple
- easy to add additional services without rejiggering the pricing structure

But there are two major complications:
- tiered unlocks don’t work with yearly subscriptions, which a lot of people enjoy in the age of subscription overload, plus yearly prepayments is [very good for startup-business](https://rein.pk/startup-finance-for-founders-part-ii-strategic-finance).
- do free sponsees unlock all services, or just a restricted selection? What if they need more?

Maybe we need to consider an alternative designed to work with annual subscriptions:

### Weird Nerd 🤓 

$50/year ($25/yr pre-alpha)

- Personal website
- Custom (sub-)domain
- Links directory (share links)
- Bookmarks manager (read-it-later)
- RSS reader (blogs inbox)
- Pastebin/notes (memos)
- (yearly subscriber gets one 🆓 invite code for one free year)

#### Extras
- Email alias/inbox 
- ActivityPub
- ATproto
- Matrix account
- Blog (PostOwl?) 


#### Rev-share for bundled apps.

For as long as we are bundling 3rd party open source apps in our core offering we will pay forward $5 of each yearly sub to those upstream services.

That money will be divided based on usage metrics mapped to some flat-fee tiers, meaning a very active Reader app might get $1.50, while infrequently used ones get $0.25 each (which is the minimum payout) and any practically unused services get $0.


#### Rev-share for extras

Every *Extra* service add-on simply costs a flat $10/yr, and it all goes directly to the service maintainer. We’ll partner with the maintainers to let them be primarily responsible for that whole service offering.

#### Free sponsees

Sponsored Writer accounts would initially be reserved for partners and contributors of the Weird project. Even later on they may only be made available to sponsees of higher trust levels. Gotta earn the privilege to speak by first doing your reading & re-sharing.

Sponsees do not get any 🆙-stream service apps. (Maybe they could be gifted by sponsors).

It's not like anyone would be barred from participating, since there are already plenty of free options to choose between among individual hosts for ActivityPub, Bluesky etc., which can all be added as external integrations to ones Weird profile.

- - -

What I like about this alternative structure is that it emphasizes Weird's primary value-proposition, which is an interweaving of multiple protocol-services into a cohesive whole. By not being forced to pick between these services individually, we relieve the user of a significant cognitive load at the onboarding stage. Instead they're free to try out different add-ons as they see fit, organically arriving at whichever app composition best suits them.

**Reply by erlend-sh** [reply]
I think we can have a smaller amount of **core apps** in each tier, and offer additional services (Weird-hosted apub account etc.) at annual-only prices, like $10 a pop.

You’d get a one month long free trial though. (Get two more months for free if you sign up for the subscription ;) )

Rev-share would change with this. Instead, it’d be something like ‘upstream service-apps get 3$ of each $10’. Might make more sense to grant them the full amount 🤔 added a (option B) accordingly.

**Comment by erlend-sh** [comment]
## Licensing

> Weird wants to serve self-hosters and cloud-subscribers on equal terms. As product developers we believe 'you become what you sell', and we want first and foremost to be software providers, not cloud providers. (Expounding blog post TBA).

While the general-purpose utilities of Weird (e.g. our graph database and any prospective protocol layer built on top of it) will be licensed as permissive open source, the Weird app will be [Fair Source](https://fair.io) licensed (very similar anyways).

https://openpath.chadwhitacre.com/2024/the-historical-case-for-fair-source/

Our version of that will entail the Polyform license plus a *countdown addendum*:
- https://polyformproject.org/licenses/noncommercial/1.0.0/
- https://writing.kemitchell.com/2023/10/24/Scheduled-Relicensing
- https://writing.kemitchell.com/2023/12/01/PolyForm-Countdown

**Reply by erlend-sh** [reply]
- https://github.com/commune-os/weird/pull/123

**Reply by aschrijver** [reply]
What are the consequences of Fair Source?

For Weird there is a single business entity that is allowed to reap commercial benefit from any contributions by community members. That is not encouraging other businesses to contribute, nor to have Weird in their techstack. Though they might contribute to improve their own experience as end-users, adding value to your benefit. Unless they consider an explicit partnership relationship the licensing of Weird seems a significant barrier for businesses to adopt and help enrich the ecosystem.

Where businesses can be encouraged to participate:

- If a plugin mechanism allowed 3rd-party plugins with different licensing.
- By becoming a true champion of the Leaf protocol as the foundation for the ecosystem.

For non-profit initiatives there may be more motivation to innovate and extend Weird, and expand its ecosystem for specific (non-profit) use cases. But given the licensing does this entail that the business entity can subsequently decide to monetize this functionality? If yes, that would hardly be fair. Claim the profit and shield it from seeping into the FOSS movement.

**Reply by aschrijver** [reply]
These [consequences](https://github.com/muni-town/weird/discussions/14#discussioncomment-11002989) may be exactly as intended, depending on the positioning of Muni Town e.g.

- Weird is our commercial end-user product.
- Weird has an official partnership network.
- Weird is open technology for non-profit use.

This position can be fine. Muni Town then mainly focuses on monetizing Weird via end-user sales, and partnerships are created along the way as opportunities arise. There might be significant FOSS contribution, which can be further stimulated. Weird is primarily developed as a product, and not fostered as a technology whose adoption and ecosystem must grow.

For potential partners and for 3rd-party technology adopters it is important where the ambition level and focus of Muni Town is. The above three bullets aren't very specific. You might be just after selling an app to end users and earn a sustainable income for two people. It would be a valid cause, but not stimulating ecosystem and contributions.

A higher ambition level would be to make explicit commitments wrt fostering a) the ecosystem and b) the technology base. Aforementioned plugin licensing scheme and championing Leaf protocol as open standard come to mind as ways to stimulate a flourishing ecosystem.

I believe in a strategy of "Leveraging the ecosystem" whereby different FOSS initiatives enrich each other, and synergetic cross-pollination and co-creation takes place. The assumption is that there's enough business potential to allow all FOSS participants to monetize, if they want to. Fair Source seems to hamper the path to monetization (at ecosystem level), not a good fit.

**Reply by zicklag** [reply]
@erlend-sh may have more thoughts on this, but here are my current thoughts.

> For non-profit initiatives there may be more motivation to innovate and extend Weird, and expand its ecosystem for specific (non-profit) use cases. But given the licensing does this entail that the business entity can subsequently decide to monetize this functionality? If yes, that would hardly be fair. 

Technically that would be possible if whomever created the Weird extension licensed their code freely under MIT or something similar, because that would allow anybody, including us, to make a profit off of it.

But if we release Weird itself under MIT, then anybody can steal Weird from _us_ the ones who developed it, and make money off of it. Amazon could take it, sell hosting at a loss subsidized by their cloud platform, and put us out of business because users would rather go with the cheaper option.

If the Weird extension was also released as non-profit, then we would have to form a partnership with the person who developed the extension if we wanted to use it in Weird to make money. If we, the Weird developers thought that was something we wanted to use on our platform, we would partner with them so that we get to use the Weird extension they developed commercially, and they get a cut of our profits in Weird, so that they are actually paid for their extension now.

---

> - Weird is our commercial end-user product.
> - Weird has an official partnership network.
> - Weird is open technology for non-profit use.

I think those points are accurate at the current stage.

> For potential partners and for 3rd-party technology adopters it is important where the ambition level and focus of Muni Town is. The above three bullets aren't very specific. 

That's definitely true.

I think that we are absolutely open to the ambition of growing into ecosystem with around Leaf, which is fully open, but our initial concern is mostly to make sure that we can make development sustainable. After that it is a matter of seeing what happens and how things play out.

Our vision includes helping give people more agency and freedom on the internet, and having an ecosystem that grows on top of a protocol and tooling that makes that easier to do for more people ( Leaf, etc. ) would be great. But that's a later phase. We don't want to sabotage our ability to even get started by concerning ourselves, _too_ much with something that won't work at all if we don't get off the ground.

Again, though, we are thinking seriously about making that a possibility. We're trying to be as pragmatic as possible, while actually getting done what needs to be done in the short term, too.

> Fair Source seems to hamper the path to monetization (at ecosystem level), not a good fit.

I'm not 100% sure, but I feel like Fair Source is important for us at the app level, and that being able to have a funded team that can work on the open source core is an advantage in this scenario.

The ecosystem health is dependent at this point on our good will, which is a dependency that will hopefully dissolve eventually, and also something that we would like to see happen, but again, we are trying to start by giving us the ability to continue to reliably spend our time in this space.

**Reply by aschrijver** [reply]
I've been trying to formulate various responses, but find I always veer into side-topics. Guess what I want to say is there's a growing group of people circling Muni Town with a "How may I fit in?" question, seeking opportunities. Who may all add value and reap collective benefits. As core team you can go whatever direction, it is all valid, because it is your initiative. Still, your decisions affect our possible role(s) here. Affects whether we'll be early adopters and invest time, energy and money, or look elsewhere.

On our [Muni Town community members] technology radar you [Weird / Leaf] are "Assess". If your business strategy now is 'focus on end-user product and sales' and objective is 'sustainable income for core team', then that gives clarity and input for further considerations. In my case it might be the decision to go to "Hold" and reassess later, when you are in a new stage on the roadmap. Weird / Leaf is then just one promising candidate on shortlists of local-first and social networking technologies.

Within the chat channels and in @erlend-sh's blog there are hints for much more to come, both in technology innovations but also socio-cultural directions. "Improving the world" type of aspirations, by finding and facilitating new ways of doing things. Ways and things that have attracted others to participate some way or other with Muni Town. How, when and where do these fit in?

A great deal of attention and elaboration has been done on the technical sides of the initiative. I wonder if it is time to fan out and focus more on other organizatorial aspects as well. Give more insight on mission, vision, business strategy, and set up + improve a product development process. I think you can be very open in the communication here, and stimulate _(spontaneous) emergence_ of an open ecosystem, without that needing your active involvement per se.

PS. Following an aforementioned "Leverage the ecosystem" strategy can be an ongoing creative exercise, an ideation process, on how to reap maximum mutual benefit from collaborative efforts and foster the emergence of an ecosystem. Find all the ways to turn onlookers into valuable co-creators.

**Reply by zicklag** [reply]
> Within the chat channels and in @erlend-sh's blog there are hints for much more to come, both in technology innovations but also socio-cultural directions. "Improving the world" type of aspirations, by finding and facilitating new ways of doing things. Ways and things that have attracted others to participate some way or other with Muni Town. How, when and where do these fit in?

I think that's a really great question, and you have a good point overall. You're the second or third person to specifically bring up our long term direction in an attempt to figure out how they want to fit into what we're doing.

Our plan was to lean into this a bit more after the 0.3 release of Weird coming up quite soon.

---

I think at this point, @erlend-sh, it might be a good idea to come up with something of a vision statement for Muni Town, and a concise overview of what we're trying to accomplish on a large, almost abstract scale.

We've shared a lot of stuff around our ideas of an "Agentic Fediverse", etc. but I think what we need is more of a "lighthouse" that shows people where we're going with this, and what our current aspirations commitment and situation is for Muni Town.

Maybe a Muni Town website. It doesn't need to be much, but something that we can keep up-to-date regarding our status and direction, and probably specific, short explanations for each of our active projects and their statuses, such as Commune, Weird, and Leaf.

---

Our biggest ambitions for the future are grand, I think. While I acknowledge that the future potential is great, though, it's hard to say with any certainty that what we are making now will grow into anything lasting. In other words, any investment in what we are making is a gamble, **but** we really do want to do all that we feasibly in good faith to help benefit the community / ecosystem that may form around our work at least as it aligns with our vision.

I for one am definitely open to thinking about more ways to more productively include community involvement for the benefit of the larger ecosystem.

**Reply by aschrijver** [reply]
> I think at this point, @erlend-sh, it might be a good idea to come up with something of a vision statement for Muni Town, and a concise overview of what we're trying to accomplish on a large, almost abstract scale.

That would be a great start. And alongside of that gather feedback of non-core team community members and lay the puzzle of how people may fit in.

I'd also like to bring up another point to consider wrt fostering emergence of an ecosystem. The potential of Weird / Leaf depends in part also on the successful introduction of new computing paradigms. First of all in the realm of "local-first apps" and secondly relating to the "open social web". There are risks if the technology base doesn't mature. And then there are big opportunities for the pioneers. Major innovation awaits and there's advantage if you can move fast in a collective with many players, take position, be ahead of the game. And an ecosystem can bring major boost to the attractiveness

> We've shared a lot of stuff around our ideas of an "Agentic Fediverse", etc. but I think what we need is more of a "lighthouse" that shows people where we're going with this, and what our current aspirations commitment and situation is for Muni Town.

Exactly. I see "agentic fediverse" more as a technology vision. What is Muni Town even?

Would it be an idea to make a clearer separation between the business entity you represent, and the collective itself including all community members (anyone who is active in the ecosystem, no member registration needed)? This way you could launch a portal focused on the collective adding their Lego blocks, with Leaf at its core. Your business being one of independent initiatives that is part of the collective, having its own web presence for e.g. Weird.

The "Leverage the ecosystem" strategy aims to reap maximum mutual benefits from each others efforts, on a continual basis. It might be literally named "Collective Development", having the Muni Town collective as its scope. And it could encompass the idea of Partnering you have. The traditional way is a "Contact Us" and then further dealings in closed partner channels. I feel that is not a good match to foster a vibrant ecosystem, though you might use this specifically for product partners relating directly to Weird.

Partnering can be made explicitly transparent. All collaboration & contributions to the ecosystem by members of a collective stir the cauldron of opportunities, and may see new partnership relations emerge. Here again it is _emergence_ that can - and in my opinion _should_ - be fostered via Collective Development.

PS. Note that imho thinking of collective members as a "community" is a fallacy, or rather what constitutes community must be well defined. The collective members operate within grassroots movements with unique social dynamics, where they rather individualistically pursue own dreams. Active participation is based on concrete incentives to collaborate and intrinsic motivation.

**Reply by erlend-sh** [reply]
Great conversation so far, and I’ll be adding to it shortly. I just need my busted-laptop replacement to arrive!

**Reply by zicklag** [reply]
> Would it be an idea to make a clearer separation between the business entity you represent, and the collective itself including all community members (anyone who is active in the ecosystem, no member registration needed)? This way you could launch a portal focused on the collective adding their Lego blocks, with Leaf at its core. Your business being one of independent initiatives that is part of the collective, having its own web presence for e.g. Weird.

I think I like this idea.

> Exactly. I see "agentic fediverse" more as a technology vision. What is Muni Town even?

I agree that agentic fediverse is a technology vision.

I'm not sure whether Muni Town is the business or the collective, if we want to draw a line there. Personally I feel like Muni Town is closer to the collective. I feel like "Weird" is the business.

Muni Town is like everybody that has kind of come together to respectfully share thoughts, collaborate, or just just hang out and watch what's happening. Muni Town is all of the people that have kind of centered around our overlapping visions of some sort.

> or rather what constitutes community must be well defined.

I have heard others who have more specific ideas of what "community" means than I tend to have when I say it. I'm not sure what is most common, but when I say "community" I usually mean anybody who ends up visibly involved

I can't remember who said it, but it made sense to me when somebody said that you can see somebody anywhere related to your project, whether because they joined the chat room or commented on a GitHub issue or whatever, then they're part of the community.

---

Anyway, definitely appreciating your perspective @aschrijver!

**Reply by aschrijver** [reply]
Yes, the Muni Town you describe is how I thought of it too.

As for community. It is probably best to always think of "community-in-context". Sometimes you can be handwavy (e.g. loosely refer to "the FOSS community") and sometimes you must define very explicitly. For instance in communal software, where Community is a core business domain.

You might use your definition of community for Muni Town collective. And set your expectations accordingly. The definition is very loose. From such definition results that you can't expect anything from the community. It includes any passers-by as they come and leave and there are no commitments. I think that corresponds effectively to a [community of interest](https://en.wikipedia.org/wiki/Community_of_interest). I would simply refer to this as the "collective" and then reserve "community" for more focused areas.

For instance for Leaf you might want to have a dedicated [community of action](https://en.wikipedia.org/wiki/Community_of_action) to evolve the protocol within the ecosystem. And you want a self-sustained community where tasks are properly delegated and people also do the chores. The actual terminology to refer this community may be different, though (might be "working group" within the collective, idk).

**Reply by aschrijver** [reply]
I found [Anytype](https://anytype.io) as a local-first p2p product with a similar licensing approach to Weird/Leaf. The company looks to be a cooperative. See [this HN comment](https://news.ycombinator.com/item?id=38797366) and the mentioned page [Our Open Philosophy](https://blog.anytype.io/our-open-philosophy/).

**Comment by zicklag** [comment]
From @erlend-sh on Discord regarding charging for top-level names, i.e. `https://a.weird.one/topLevelName`, which can be a project/user/org/whatever:

> Let’s say every account gets one top level name for free. You get 10 with a yearly subscription.

**Comment by erlend-sh** [comment]
What do y’all think of this offer?

**Buy now ($25) 💵 get two subsequent years for free** 🆓 🆓 

It’s like a ‘lifetime subscription’ middle-ground, available only to earliest adopters. We need to be signing up 100 additional yearly subscribers per month anyway for baseline sustainability, so it doesn’t make much of a difference to our long term financials if the first few batches (months) of paying subscribers get a much sweetened deal.

These early adopters are so early that we will also make it clear for them: ‘our existence past the next 6 months isn’t yet guaranteed though, fyi 🤷‍♂️ ‘

**Reply by zicklag** [reply]
That sounds reasonable to me. :+1:
